server/service/src/boajs/utils.rs

```diff
@@ -64,3 +64,55 @@ pub(super) fn get_serde_argument<D: DeserializeOwned>(
 
     closure().map_err(std_error_to_js_error)
 }
+
+#[derive(Error, Debug)]
+pub enum ExecuteGraphQlError {
+    #[error(transparent)]
+    Serde(#[from] serde_json::Error),
+    #[error("GraphQL error: {0}")]
+    Graphql(String),
+}
+
+#[async_trait::async_trait]
+pub trait ExecuteGraphql: 'static + Send + Sync {
+    async fn execute_graphql(
+        &self,
+        override_user_id: &str,
+        query: &str,
+        variables: serde_json::Value,
+    ) -> Result<serde_json::Value, ExecuteGraphQlError>;
+}
+
+/// Executes an async function in a blocking manner from within an async or sync context.
+/// Required to run async code, like GraphQL queries, from boa js sync context.
+/// Functionality here allows for two examples, one for actix_web server and one for main runtime.
+/// Can be seen in action in open-msupply-plugins repository, through graphql query and message processor
+pub fn do_async_blocking<T, F>(handle: Arc<Handle>, f: F) -> T
```

Comment:
I've tested this in many ways, one method doesn't work for both actix_web server and for normal runtime things, like processors, it's quite hard to do a normal unit test for this, however this should be tested within plugin repo against docker build of omSupply, on my todo